File: irmap.py
def irmap(query_id, relevant_doc_ids, query_file):
    path = query_file[0:query_file.rfind('\\')]
    relevant_doc_ids = [int(i) for i in relevant_doc_ids]
    relevance_file = path+"\\relevance."+query_id #to fetch relevance values from relavance file4
    relevance = []
    f =  open(relevance_file, "r")
    content = f.read()
    split_content = content.splitlines()
    for i in range(len(split_content)): #searching in relevance file for the relevance of each document
        if i in relevant_doc_ids:
            relevance.append(int(split_content[i]))
   # relevance = [1,0, 2]
    precision = []

    relCount = 0
    precisionValue = 0
    for i in range(len(relevance)):
        relCount = relCount + relevance[i]
        print("Total relevance at " + str(i+1) + ": " + str(relCount))
        precisionValue = relCount/(i+1)
        print("Total precision at " + str(i+1) + ": " + str(precisionValue))
        precision.append(precisionValue)
    sum = 0
    for i in precision:
        sum = sum + i

    try:
        map = sum/(len(precision))
    except:
        print("*** Relevance files empty ***")
        map = 0


    print("The MAP value is: " + str(map))

    return map

File: IRMAP.java
import java.util.*;

public class IRMAP{

	public static void main(String [] args){
		ArrayList<Double> relevance = new ArrayList<>(); // stops at the N value
		ArrayList<Double> precision = new ArrayList<>(); 
		// Add relevance values
		Scanner keyboard = new Scanner(System.in);
		System.out.print("Enter N value: "); 
		int N = keyboard.nextInt(); 
		System.out.println("Enter N relevance values:");
		// add revevance values to arraylist
		for(int a = 0; a < N; a++){
			int rel = keyboard.nextInt(); // should be 0, 1 or 2
			double actualRel = rel/2.0;
			System.out.println(actualRel);
			relevance.add(actualRel);
		}

		// get precisions at each point
		double relCount = 0; 	
		double precisionValue = 0;
		for(int a = 0; a < N; a++){
			relCount = relCount + relevance.get(a);
			System.out.println("Total relevance at " + (a+1) + ": " + relCount);
			
			precisionValue = relCount/(a+1);
			System.out.println("Total precision at " + (a+1) + ": " +precisionValue);
			
			precision.add(precisionValue);			
		}

		// find the sum and average
		double sum = 0;	
		for(double element : precision){
			sum = sum + element;
		}
		double map = sum/(precision.size());
		System.out.println("The MAP value " + map);
	}
}
